Cython/Compiler/Nodes.py

```diff
@@ -4612,44 +4588,35 @@ def analyse_declarations(self, env):
                 env.add_imported_module(self.module)
 
-        if self.base_class_name:
-            if self.base_class_module:
-                base_class_scope = env.find_imported_module(self.base_class_module.split('.'), self.pos)
-                if not base_class_scope:
-                    error(self.pos, "'%s' is not a cimported module" % self.base_class_module)
-                    return
-            else:
-                base_class_scope = env
-            if self.base_class_name == 'object':
-                # extension classes are special and don't need to inherit from object
-                if base_class_scope is None or base_class_scope.lookup('object') is None:
-                    self.base_class_name = None
-                    self.base_class_module = None
-                    base_class_scope = None
-            if base_class_scope:
-                base_class_entry = base_class_scope.find(self.base_class_name, self.pos)
-                if base_class_entry:
-                    if not base_class_entry.is_type:
-                        error(self.pos, "'%s' is not a type name" % self.base_class_name)
-                    elif not base_class_entry.type.is_extension_type and \
-                             not (base_class_entry.type.is_builtin_type and
-                                  base_class_entry.type.objstruct_cname):
-                        error(self.pos, "'%s' is not an extension type" % self.base_class_name)
-                    elif not base_class_entry.type.is_complete():
-                        error(self.pos, "Base class '%s' of type '%s' is incomplete" % (
-                            self.base_class_name, self.class_name))
-                    elif base_class_entry.type.scope and base_class_entry.type.scope.directives and \
-                             base_class_entry.type.is_final_type:
-                        error(self.pos, "Base class '%s' of type '%s' is final" % (
-                            self.base_class_name, self.class_name))
-                    elif base_class_entry.type.is_builtin_type and \
-                             base_class_entry.type.name in ('tuple', 'str', 'bytes'):
-                        error(self.pos, "inheritance from PyVarObject types like '%s' is not currently supported"
-                              % base_class_entry.type.name)
-                    else:
-                        self.base_type = base_class_entry.type
-                if env.directives.get('freelist', 0) > 0:
-                    warning(self.pos, "freelists cannot be used on subtypes, only the base class can manage them", 1)
+        if self.bases.args:
+          base = self.bases.args[0]
+          base_type = base.analyse_as_type(env)
+          if base_type in (PyrexTypes.c_int_type, PyrexTypes.c_long_type, PyrexTypes.c_float_type):
+              # Use the Python rather than C variant of these types.
+              base_type = env.lookup(str(base_type)).type
```

Comment:
I'd rather call `sign_and_name()` here instead of relying on the output of `str()`.